Libraries/LibCore/MeasureTime.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2025, Tim Ledbetter <tim.ledbetter@ladybird.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "MeasureTime.h"
+
+namespace Core {
+
+HashMap<ByteString, TimingInfo> g_timing_info_table;
```

Comment:
couldn't this be replaced by function static instances?
init would be slightly longer line-wise though*

or is there a plan to expose this for a togglable at-exit print of the stats?

*:
```cpp
static auto __timing_stats_##name = TimingInfo(name, print_every_n);
auto measure_time_timer_##name = Core::ElapsedTimer::start_new(Core::TimerType::Precise);
ScopeGuard measure_time_guard_##name = [&] { 
    auto elapsed_nanoseconds = measure_time_timer_##name.elapsed_time().to_nanoseconds();
    __timing_stats_##name.add_run(elapsed_nanoseconds);
};
```

Reply:
also the `##name` could be replaced by `##__COUNTER__` but its likely not thaat useful to have timers of the same name in the same scope